.github/workflows/uploadWheels.yml

```diff
@@ -51,16 +53,24 @@ jobs:
           CIBW_BUILD_FRONTEND: build
           SETUPTOOLS_SCM_DEBUG: True
 
-      - name: Upload wheels (Non-Tag)
+      - name: Upload wheels (Tag)
+        uses: pypa/gh-action-pypi-publish@release/v1
+        if: github.ref_type == 'tag'
+        with:
+          password: ${{ secrets.PYPI_CIRCT_API_TOKEN }}
+          packages-dir: wheelhouse/
+          verify-metadata: false
+
+      - name: Upload wheels (Non-Tag and Non-Nightly)
         uses: actions/upload-artifact@v3
-        if: github.ref_type != 'tag'
+        if: github.ref_type != 'tag' && github.event_name != 'cron'
         with:
           path: ./wheelhouse/*.whl
           retention-days: 7
 
-      - name: Upload wheels (Tag)
+      - name: Upload wheels (Non-Tag and Nightly)
         uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
@mikeurbach @teqdruid how do we version this to be made into a dev release on pypi?

Reply:
If the versioning works like the pycde versioning scheme, breaking it out isn't necessary. The versioning lib automatically detects the most recent tag in the commit's history. If the commit *is* the tag, it uses that. If not, it appends `.dev<numberOfCommitsSinceTag>`.

Reply:
So you really just need the schedule addition above. Am I correct @mikeurbach ?

Reply:
We do [use](https://github.com/llvm/circt/blob/5dffe3a5bf59afe653a3ea3c23875f485ec61914/lib/Bindings/Python/pyproject.toml#L14) setuptools_scm, so the versioning should work as expected. But I think it makes sense to have three steps:
* one for when there is a tag (which is run on the release event trigger) and uploads the tag version to PyPI
* one for when the event is the cron, which uploads the dev version to PyPI
* one for anything else (useful for testing the workflow), which uploads artifacts to the job for inspection

I think it would make sense to use:

```
- name: Upload wheels (Tag)
   if: github.ref_type == 'tag'

- name: Upload wheels (Nightly)
   if: github.event_name == 'cron'

- name: Upload wheels (Workflow Dispatch)
   if: github.ref_type != 'tag' && github.event_name != 'cron'
```

What do you think @teqdruid @rsetaluri ?

Reply:
What would be the actions? My point is, how would they differ?

Reply:
Oh I see, yes we can combine the steps that push to pypi, something like:

```
- name: Upload wheels (Tag or Nightly)
   if: github.ref_type == 'tag' || github.event_name == 'cron'
```

Reply:
Yeah. I was assuming that you'd also want to push to pypi on a workflow_dispatch, but perhaps not...

Reply:
Yeah, I think we should only push to pypi on automated builds. If someone is testing the workflow itself or something, I don't think we need to push new versions.

Reply:
agree with the two modes. In the case of `!(tag | cron)`, we can just download artifacts from github actions